ibis/expr/tests/test_newrels.py

```diff
@@ -1314,3 +1314,45 @@ def test_join_method_docstrings():
         join_method = getattr(joined, method)
         table_method = getattr(t1, method)
         assert join_method.__doc__ == table_method.__doc__
+
+
+def test_join_with_compound_predicate():
+    t1 = ibis.table(name="t", schema={"a": "string", "b": "string"})
+    t2 = t1.view()
```

Comment:
Without this it should fail, but that is okay since we are unable to figure out the sides for nested predicates.

Reply:
Sure, but the goal here was to have a test case for the specific problem. The test will fail as written without the change to the deref algorithm